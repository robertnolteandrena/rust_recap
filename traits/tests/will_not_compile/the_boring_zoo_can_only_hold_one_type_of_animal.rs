use spectral::assert_that;
use traits_lib::{BoringZoo, Goat, HasName, Sheep};

fn boring_zoo_can_only_hold_one_type_of_animal() {
    let goat = Goat;
    let sheep = Sheep;
    let boring_zoo = BoringZoo {
        animals: vec![goat, sheep],
    };
    let boring_zoo_name = boring_zoo.name();
    assert_that!(boring_zoo_name.as_ref()).is_equal_to("BoringZoo of: Goat Sheep");
}
fn main() {}
