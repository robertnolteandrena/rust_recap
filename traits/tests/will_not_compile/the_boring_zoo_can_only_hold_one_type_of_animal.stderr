error[E0308]: mismatched types
 --> tests/will_not_compile/the_boring_zoo_can_only_hold_one_type_of_animal.rs:8:29
  |
8 |         animals: vec![goat, sheep],
  |                             ^^^^^ expected `Goat`, found `Sheep`
